kratos/elements/distance_calculation_flux_based_element.h

```diff
@@ -0,0 +1,359 @@
+    ///@{
+
+    /**
+     * @brief Assembles the local system for a laplacian, no coeffs
+     * @return A laplacian systes
+     */
+    // void CalculateLaplacianSystem(
+    //     MatrixType &rLeftHandSideMatrix,
+    //     VectorType &rRightHandSideVector,
+    //     const ProcessInfo &rCurrentProcessInfo);
+
+    /**
+     * @brief Assembles the local system to get a gradient in the direction of the flow
+     * @return A transient diffusion system whose solution can be used to construct a velocity field
+     * this velocity will then be used to compute the flowlenght/filltime in a second step
+     */
+    void CalculatePotentialFlowSystem(
+        MatrixType &rLeftHandSideMatrix,
+        VectorType &rRightHandSideVector,
+        const ProcessInfo &rCurrentProcessInfo);
+
+    /**
+     * @brief Assembles the local system to get a the flowlength/filltime
+     * @return A pure convection + sorce term system that solves the flowlength/filltime.
+     * For the FlowLength, the assembled source term is the same as |vel|,
+     * therefore obtaining a gradient if 1 in the direction of the velocity
+     * For the FillTime, the assembled source term is 1.0
+     * therefore the gradient is 1/|vel| in the directiion of the velocity
+     */
+    void CalculateDistanceSystem(
+        MatrixType &rLeftHandSideMatrix,
+        VectorType &rRightHandSideVector,
+        const ProcessInfo &rCurrentProcessInfo);
+        
+    //computes characteristic length
+    double ComputeH(BoundedMatrix<double,NumNodes, TDim>& DN_DX);
+
+    void CalculateGaussPointsData(  const GeometryType& rGeometry,
+                                    BoundedVector<double,TNumNodes> &rGaussWeights,
+                                    BoundedMatrix<double,TNumNodes,TNumNodes> &rNContainer,
+                                    array_1d<BoundedMatrix<double, TNumNodes, TDim>, TNumNodes>& rDN_DXContainer 
+                                 );
+    
+    void GetSimplexShapeFunctionsOnGauss(BoundedMatrix<double,TNumNodes, TNumNodes>& Ncontainer);
+
+
+    ///@}
+    ///@name Protected  Access
+    ///@{
+
+    ///@}
+    ///@name Protected Inquiry
+    ///@{
+
+    ///@}
+    ///@name Protected LifeCycle
+    ///@{
+
+    ///@}
+    ///@name Static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Member Variables
+    ///@{
+
+    ///@}
+    ///@name Serialization
+    ///@{
+    
+
+
+    ///@}
+private:
+
+    friend class Serializer;
+
+    void save(Serializer& rSerializer) const override
+    {
+        KRATOS_SERIALIZE_SAVE_BASE_CLASS(rSerializer, Element );
+    }
+
+    void load(Serializer& rSerializer) override
+    {
+        KRATOS_SERIALIZE_LOAD_BASE_CLASS(rSerializer, Element);
+    }
+
+    /// Assignment operator.
+    DistanceCalculationFluxBasedElement& operator=(DistanceCalculationFluxBasedElement const& rOther);
+
+    /// Copy constructor.
+    DistanceCalculationFluxBasedElement(DistanceCalculationFluxBasedElement const& rOther);
```

Comment:
```suggestion
    /// Copy constructor.
    DistanceCalculationFluxBasedElement(DistanceCalculationFluxBasedElement const& rOther) = delete;
```
And move it to the life cycle section where it actually belongs.

Reply:
ok